### PR TITLE
Using Subjects within Streams

### PR DESCRIPTION
**Pull request by JeanPierreNell**
-enable user to make use of subjects inside streams
-backwards compatible

### CODE REVIEW

File: src/services/jetstreamService.ts [hunk 1]
```diff
@@ -144,11 +144,13 @@ async function validateEnvironment(): Promise<void> {
 
 async function createConsumer(functionName: string, jsm: JetStreamManager, consumerStreamName: string): Promise<void> {
   await createStream(jsm, consumerStreamName, startupConfig.streamSubject ? startupConfig.streamSubject : undefined);
+  const streamSubjects = startupConfig.streamSubject ? startupConfig.streamSubject.split(',') : [startupConfig.consumerStreamName];
```
Comment (johanfol) [inline src/services/jetstreamService.ts:147]:
```suggestion
  const streamSubjects = startupConfig.streamSubject ? startupConfig.streamSubject.split(',') : startupConfig.consumerStreamName.split(',');
```

File: src/services/jetstreamService.ts [hunk 2]
```diff
@@ -190,9 +192,11 @@ async function createStream(jsm: JetStreamManager, streamName: string, subjectNa
  *
  * @return {*}  {Promise<void>}
  */
-export async function handleResponse(response: string, subject?: string): Promise<void> {
+export async function handleResponse(response: string, subject: string[]): Promise<void> {
   const sc = StringCodec();
-  if (producerStreamName) await js.publish(subject ?? producerStreamName, sc.encode(response));
+  for (const sub of subject) {
```
Comment (johanfol) [inline src/services/jetstreamService.ts:197]: Do a promise.all here instead of awaiting each publish please.

**Review by johanfol — CHANGES_REQUESTED** [review]

File: src/services/jetstreamService.ts (hunk 2 above)
Comment (JeanPierreNell) [reply src/services/jetstreamService.ts]: Done

File: src/services/jetstreamService.ts [hunk 3]
```diff
@@ -143,33 +143,43 @@ async function validateEnvironment(): Promise<void> {
...
   await jsm.streams.find(streamName).then(
-    (s) => {
-      logger.log('Producer stream already exists');
+    async (stream) => {
+      logger.log(`Stream: ${streamName} already exists.`);
+
+      if (subjectName) {
+        logger.log(`Adding subject: ${subjectName} to stream: ${streamName}`);
+        const streamInfo = await jsm.streams.info(stream);
+        streamInfo.config.subjects?.push(subjectName);
```
Comment (johanfol) [inline src/services/jetstreamService.ts:167]: If there is no subjects, this push will not add any subjects? So don't you need an if check to say if no subjects, create?

**Review by johanfol — CHANGES_REQUESTED** [review]

**Pull request closed without merging**